areas/foundry/src/AzureMcp.Foundry/Services/FoundryService.cs

```diff
@@ -164,15 +166,20 @@ public async Task<ModelDeploymentResult> DeployModel(string deploymentName, stri
 
         try
         {
-            ArmClient armClient = await CreateArmClientAsync(null, retryPolicy);
-
-            var subscription =
-                armClient.GetSubscriptionResource(SubscriptionResource.CreateResourceIdentifier(subscriptionId));
-            var resourceGroupResource = await subscription.GetResourceGroupAsync(resourceGroup);
-
-            var cognitiveServicesAccounts = resourceGroupResource.Value.GetCognitiveServicesAccounts();
-            var cognitiveServicesAccount = await cognitiveServicesAccounts.GetAsync(azureAiServicesName);
-
+            // Create ArmClient for deployments
+            var options = new ArmClientOptions();
+            options.SetApiVersion("Microsoft.CognitiveServices/accounts/deployments", "2025-06-01");
+            ArmClient armClient = await CreateArmClientAsync(null, retryPolicy, options);
+
+            // Retrieve the Cognitive Services account
+            var genericResources = armClient.GetGenericResources();
+            var cognitiveServicesAccount = await genericResources.GetAsync(
+                new ResourceIdentifier($"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.CognitiveServices/accounts/{azureAiServicesName}"));
+            if (!cognitiveServicesAccount.Value.HasData)
+                throw new InvalidOperationException($"Services account '{azureAiServicesName}' not found in resource group '{resourceGroup}'.");
+
+            // Prepare data for the deployment
+            ResourceIdentifier id = new ResourceIdentifier($"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.CognitiveServices/accounts/{azureAiServicesName}/deployments/{deploymentName}");
```

Comment:
Seems like this type of code will be common for the RM lib support.  Can we refactor this to think about future?  I'm thinking we have helpers so it's more plug and play.